src/utils/json-editor-parser.ts

```diff
@@ -89,11 +89,16 @@ const relationships = (xs: { id: string; children: never[] }[]) => {
   ]);
 };
 
+export const flattenTree = tree => {

```

Comment:
I've needed to break in two the function parser on line 97.
- extractTree will take as input the JSON data, and outputs the node tree that contains the `id` and `children` keys. This is the main data structure to work on.
- flattenTree is the function that will take as an input the extracted tree produced by extractTree, and will flatten it to an array. That array can be later on used to split the nodes and edges. I am missing some tests here, because on the test files you can take a look at the input/output data structures and have a better idea of what each function does, without need to read the source code of them. Take a look at this [branch's folder](https://github.com/KarlHeitmann/jsonvisio.com/tree/basic_test/src/components/__tests__/utils/json-editor-parser) to see the tests